src/main/java/me/kixstar/dungeons/dungeons/Loot.java

```diff
@@ -15,8 +16,25 @@
                 .collect(Collectors.toList());
     }
 
-    public static List<ItemStack> generateLoot(LootConfiguration configuration) {
-        return null;
+    private static List<ItemStack> generateLoot(LootConfiguration configuration) {
+        List<ItemStack> finalItemStackList = new ArrayList<>();
+
+        if (Math.random() <= configuration.getChance()) {
+            int itemAmount = configuration.getItemStack().getAmount();
+            int itemsForStack;
+            int maxStackSize = configuration.getMaxStackSize();
+
+            while (itemAmount > 0) {
+                itemsForStack = itemAmount - maxStackSize >= 0 ? maxStackSize : itemAmount;
```

Comment:
You can use [Math.min](https://www.tutorialspoint.com/java/lang/math_min_int.htm) instead.